testnet/stacks-node/src/burnchains/bitcoin_regtest_controller.rs

```diff
@@ -787,50 +917,169 @@ impl BitcoinRegtestController {
 
         for commit_to in payload.commit_outs.iter() {
             tx.output
-                .push(commit_to.to_bitcoin_tx_out(value_per_transfer));
+                .push(commit_to.to_bitcoin_tx_out(estimated_fees.amount_per_output()));
         }
 
+        let fee_rate = estimated_fees.fee_rate;
         self.finalize_tx(
             &mut tx,
-            budget_for_outputs,
-            btc_miner_fee + rbf_fee,
-            utxos,
+            estimated_fees.total_spent_in_outputs(),
+            estimated_fees.spent_in_attempts,
+            estimated_fees.min_tx_size(),
+            fee_rate,
+            &mut utxos,
             signer,
-            attempt,
         )?;
 
-        increment_btc_ops_sent_counter();
+        let serialized_tx = SerializedTx::new(tx.clone());
+
+        let tx_size = serialized_tx.bytes.len() as u64;
+        estimated_fees.register_replacement(tx_size);
+        let mut txid = tx.txid().as_bytes().to_vec();
```

Comment:
Might want to consider adding a constructor to our `Txid` that takes a `rust-bitcoin` txid value and does this low-level byte manipulation.